src/utils/jsonwriter.cpp

```diff
@@ -391,13 +392,13 @@ void AsmParser::JsonWriter::JsonWriter::write()
 AsmParser::DebugJsonWriter::DebugJsonWriter(std::ostream &out,
                                             const std::vector<std::unique_ptr<asm_line_v>> &lines,
                                             const std::vector<asm_labelpair> &labels,
-                                            const Filter filter,
-                                            const std::unordered_map<std::string_view, std::unordered_set<std::string_view>> used_labels,
-                                            const std::unordered_map<std::string_view, std::unordered_set<std::string_view>> used_weak_labels,
-                                            const std::unordered_map<std::string_view, std::string_view> aliased_labels,
-                                            const std::unordered_map<std::string_view, std::unordered_set<std::string_view>> used_data_labels)
-: AsmParser::JsonWriter::JsonWriter(out, lines, labels, filter), used_labels(used_labels),
-  used_weak_labels(used_weak_labels), aliased_labels(aliased_labels), used_data_labels(used_data_labels)
+                                            const Filter &filter,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_labels,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_weak_labels,
+                                            std::unordered_map<std::string_view, std::string_view>  aliased_labels,
+                                            std::unordered_map<std::string_view, std::unordered_set<std::string_view>>  used_data_labels)
+: AsmParser::JsonWriter::JsonWriter(out, lines, labels, filter), used_labels(std::move(used_labels)),
```

Comment:
I'm not sure these moves are wise. When `-debugdump` is passed, both DebugJsonWriter and JsonWriter are instantiated with the same variables (main.cpp line 101), but they're moved into this and then reused elsewhere...

Reply:
Oh interesting! clang-tidy suggested them, and normally is pretty good about those things! Good spot though!